api/topic.go

```diff
@@ -19,18 +19,51 @@ func (api *API) GetTopicHandler(w http.ResponseWriter, req *http.Request) {
 		handlers.CollectionID.Header(): hColID,
 		"request_id":                   ctx.Value(dprequest.RequestIdKey),
 		"topic_id":                     id,
+		"function":                     "getTopicHandler",
 	}
 
 	// get topic from mongoDB by id
-	topic, err := api.mongoDB.GetTopic(req.Context(), id)
+	topic, err := api.dataStore.Backend.GetTopic(req.Context(), id)
 	if err != nil {
 		handleError(ctx, w, err, logdata)
 		return
 	}
 
-	if err := WriteJSONBody(ctx, topic, w, logdata); err != nil {
+	// Ensure the sub document has the main document ID
+	topic.Current.ID = topic.ID
+
+	// User is not authenticated and hence has only access to current sub document
+	if err := WriteJSONBody(ctx, topic.Current, w, logdata); err != nil {
```

Comment:
I think there need to be an error log here - if we failed to write the response body that should be a 500 status but if we can't write it to the response we will 100% want to know the details of what went wrong.

Reply:
WriteJSONBody calls handleError internally in two places, so the error is handled.